test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -51,17 +52,15 @@ def test_server_error
       13 => 'Internal',
       14 => 'Unavailable' }.each_with_index do |(code, message), index|
       exception_count = 0
-      setup_logging_stubs(true, code, message) do
-        d = create_driver(USE_GRPC_CONFIG, 'test',
-                          GRPCLoggingMockFailingService.rpc_stub_class)
+      setup_logging_stubs do
+        d = create_driver(
+          USE_GRPC_CONFIG, 'test',
+          GRPCLoggingMockFailingService.new(code, message, @failed_attempts))
         # The API client should retry this once, then throw an exception which
         # gets propagated through the plugin
         d.emit('message' => log_entry(0))
         begin
           d.run
-        rescue GRPC::Cancelled
-          # No need to check the message -- we already know the code.
```

Comment:
Was this code wrong, or does it no longer apply?

Reply:
Turned out that `GRPC::Cancelled` is a type of `GRPC::BadStatus`. So it was unnecessary to treat that one separately.